src/ra_server.erl

```diff
@@ -67,7 +68,10 @@
       aux_state => term(),
       condition => ra_await_condition_fun(),
       condition_timeout_effects => [ra_effect()],
-      pre_vote_token => reference()
+      pre_vote_token => reference(),
+      query_index => non_neg_integer(),
```

Comment:
Are these new fields optional or should they use `:=` - i.e. be required to be in the state map